modules/publisher/views.py

```diff
@@ -26,37 +27,42 @@ def refresh_redirect(path):
     return flask.redirect(path)
 
 
+def _handle_errors(api_error: ApiError):
+    if type(api_error) is ApiTimeoutError:
+        return flask.abort(504, str(api_error))
+    elif type(api_error) is MacaroonRefreshRequired:
+        return refresh_redirect(
+            flask.request.path
+        )
+    else:
+        return flask.abort(502, str(api_error))
+
+
+def _handle_error_list(errors):
+    codes = []
+    for error in errors:
+        if error['code'] == 'user-not-ready':
+            if 'has not signed agreement' in error['message']:
```

Comment:
This seems a little flaky. Could you create a list of internal error codes to ref here instead of checking the error messages? This would fail and be difficult to catch in QA if someone changes the message. Also, in the future, we may need to localise the messages which will make this logic difficult.

Reply:
@anthonydillon unfortunatly we can't really do in another way. This errors code is not unique... We discussed about it with @WillMoggridge I added a comment and an issue so that we move this piece of code in the API module since this is a response handled in the view module.

Reply:
Yep, we had similar issue in BSI - error needs to be identified by message: https://github.com/canonical-websites/build.snapcraft.io/blob/master/src/common/actions/auth-store.js#L248